KEXPPower/Public API/KEXPPower.swift

```diff
@@ -8,98 +8,68 @@
+    }
 
-    static var kexpBaseURL: String!
-    static var playURL = URL(string: kexpBaseURL + "/v2/plays")!
-    static var showURL = URL(string: kexpBaseURL + "/v2/shows")!
-    static var showStartURL = URL(string: kexpBaseURL + "/get_show_start/")!
-    static var streamingURL = URL(string: kexpBaseURL + "/get_streaming_url")!
+    private var kexpBaseURL: String!
 
     /// Configure KEXPPower
     /// - Parameters:
     ///   - kexpBaseURL: Base URL for making network requests
-    ///   - configurationURL: URL for retrieving app configuration file
-    ///   - availableStreams: Available live streams the user can choose from
-    ///   - archiveStreams: Available archive streams the user can choose from
     ///   - selectedArchiveBitRate: User's selected archive bitrate
-    public func setup(
-        kexpBaseURL: String,
-        configurationURL: URL? = nil,
-        availableStreams: [AvailableStream],
-        archiveStreams: [ArchiveStream]? = nil,
-        selectedArchiveBitRate: ArchiveBitRate)
-    {
-        KEXPPower.kexpBaseURL = kexpBaseURL
-        KEXPPower.configurationURL = configurationURL
-        KEXPPower.availableStreams = availableStreams.map({ (stream) -> AvailableStream in
-            return AvailableStream(streamName: stream.streamName,
-                                   streamURL: KEXPPower.appendListenerId(toURL: stream.streamURL))
-        })
-        KEXPPower.archiveStreams = archiveStreams?.map({ (stream) -> ArchiveStream in
-            return ArchiveStream(archiveBitRate: stream.archiveBitRate,
-                                 streamURL: KEXPPower.appendListenerId(toURL: stream.streamURL))
-        })
-        KEXPPower.selectedArchiveBitRate = selectedArchiveBitRate
-    }
-
-    static func appendListenerId(toURL url: URL) -> URL {
-        var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false)
-        urlComponents?.queryItems = [
-            URLQueryItem(name: "listenerId", value: listenerId.uuidString)
-        ]
-        if let urlWithListenerId = urlComponents?.url {
-            return urlWithListenerId
-        }
-        return url
+    public func setup(kexpBaseURL: String, selectedArchiveBitRate: ArchiveBitRate) {
+        self.kexpBaseURL = kexpBaseURL
+        self.selectedArchiveBitRate = selectedArchiveBitRate
     }
-
-    static var streamURL: URL {
-        guard
-            let defaultStreamURL = availableStreams?.first?.streamURL
-        else {
-            assertionFailure("Gotta have a streamURL dude.")
-            
-            return URL(string: "")!
-        }
+    
+    public var streamURL: URL {
+        let availableStreams = AvailableStreams(with: KEXPPower.sharedInstance.listenerId)
         
-        return defaultStreamURL
+        switch KEXPPower.sharedInstance.selectedArchiveBitRate {
+        case .thirtyTwo:
+            return availableStreams.livePlayback[ArchiveBitRate.thirtyTwo.rawValue]
```

Comment:
IMO this switch feels a bit fragile to me--selecting the correct stream is dependent on the ordering of the `livePlayback` list in the `AvailableStreams` instance & the indices used to select an item from that list are also dependent on the ordering of the `ArchiveBitRate` enum.  

Is it possible to instead do a more deliberate lookup of the stream url (in the `AvailableStreams` class) using the actual value of the selected bitrate, rather than by the index? Or, more specifically, just a `struct` or an `enum` where constants are defined for each of the particular streams?

Reply:
Cleaned this up a bit